src/grpc/rpc_server.cpp

```diff
@@ -155,11 +155,23 @@ bool GrpcServer::run_generic_handler_cb(const std::string& rpc_name, boost::intr
             // respond immediately
             return true;
         }
-        cb = it->second;
+        cb = it->second.first;
     }
     return cb(rpc_data);
 }
 
+void GrpcServer::run_generic_completion_cb(const std::string& rpc_name,
+                                           boost::intrusive_ptr< GenericRpcData >& rpc_data) {
+    generic_rpc_completed_cb_t cb;
+    {
+        std::shared_lock< std::shared_mutex > lock(m_generic_rpc_registry_mtx);
+        auto it = m_generic_rpc_registry.find(rpc_name);
+        if (it == m_generic_rpc_registry.end() || !(it->second.second)) { return; }
+        cb = it->second.second;
+    }
+    cb(rpc_data);
```

Comment:
cb could be nullptr right?

Reply:
Yes, it can be nullptr. It is checked inside the if condition in line 169

Reply:
how about remove line:169, 
if (cb) { cb(rpc_data); }